### ISSUE TITLE
Named type parameter

### ISSUE THREAD

**Issue by tinganho** [Microsoft/TypeScript#5053]
A spin-off idea from #2175. If a class has many type parameters and let say that default type parameter gets landed. And If I have a class with three type parameters like below:

```typescript
class CustomComponent<P, S, E> extends React.Component<P = {}, S = {}, E = {}> {
}
```

Now, I just want to assign a type to one of the type parameter `S`:

```typescript
 new CustomComponent<{}, { a: string }, {}>;
```
The above solution is quite redundant since you need to assign a type to the type parameters `P` and `E` too. When in fact, you just want to assign a type to the type parameter `S`.

I suggest `named type parameter` to solve this problem:

```typescript
 new CustomComponent<S = { a: string }>;
```
In the above solution, I don't need to assign any type to the type parameters `P` and `E`.

**Comment by mhegazy** [comment]
this looks like a syntax proposal for the implementation of #2175. if you agree, i would keep it in #2175, instead of a separate issue.

**Comment by tinganho** [comment]
OK I its fine.

**Comment by mhegazy** [comment]
thanks!

**Comment by tinganho** [comment]
@mhegazy shouldn't this issue be reopnened, since the PR #13487 is mentioning that it fixes #2175, but does not include named type argument?

**Comment by tinganho** [comment]
@mhegazy can you revisit this? I think you miss a lot if you just have #2175 and not this one.
